### ISSUE TITLE
Prometheus Agent: consider never moving samples to checkpoint

### ISSUE THREAD

**Issue by rfratto** [prometheus/prometheus#9848]
Prometheus Agent will currently move samples to checkpoints if, during GC, the timestamp of those samples are behind the lowest sent timestamp across all remote_write endpoints. The intent is that samples shouldn't be deleted before they are sent. 

However, remote_write will never read samples from checkpoints. We should consider always dropping samples (but not series) when creating a new checkpoint for the Prometheus Agent WAL. 

As an aside: @gouthamve pointed out that remote_write doesn't currently any existing samples at all from an existing WAL, which makes having a WAL seem a little pointless. I know that I've heard talk from @csmarchbanks and @cstyan of using some kind of marker so remote_write can re-start from a marker, but I don't know what the status of that is. Until we have that, I do agree that the Prometheus Agent's (and grafana/agent's) WAL isn't really providing any benefits to the user.

**Comment by gouthamve** [comment]
I wouldn't say its not providing any benefits. It is, when the remote is down. 

Having said that, it doesn't help when the agent restarts. We basically restart onto a clean slate, and can save a lot of time during restarts by just deleting the WAL.

**Comment by rfratto** [comment]
> I wouldn't say its not providing any benefits. It is, when the remote is down.

It seems like the most you'd get would be using less memory when remote_write is backed up by using the disk as a buffer. Am I missing anything else?

**Comment by roidelapluie** [comment]
Yes because the agent should ideally work with a multiple hours outage.

**Comment by cstyan** [comment]
I think even with the remote write checkpoint/marker we don't need samples in the WAL checkpoint for the agent.

**Comment by rfratto** [comment]
I'm pretty confident that's true. We've also noticed that the WAL replay for loading in series can take _forever,_ partially because of how many samples might exist in checkpoints. I think there's a few changes we can do here:

1. Don't store samples in checkpoints, since they're never read by remote_write 
2. During replay, initialize in-memory series with a lastTs of `math.MinInt64` to indicate that it's new. This is opposed to the current initialization of 0 with lastTs being eventually set to the most recent TS in the WAL for that series. 
3. Given 2, don't decode samples while replaying, which will massively improve WAL replay speed

**Comment by bboreham** [comment]
Hello from the bug scrub!

Since this issue was opened, a new `sample_age_limit` was added to remote-write config; I would argue that this could be applied to checkpointing to achieve the intent here, without adding an additional arbitrary cut-off.

(I am assuming at some time the WAL _will_ be used to replay data after a restart, which is as noted not the case today)
